server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java

```diff
@@ -402,12 +403,14 @@ public GlobalSession readSession(String xid) {
             Pipeline pipelined = jedis.pipelined();
             statusKeys.stream().forEach(statusKey -> pipelined.lrange(statusKey, 0, -1));
             List<List<String>> list = (List<List<String>>)(List)pipelined.syncAndReturnAll();
+            pipelined.close();
             List<GlobalSession> globalSessions = Collections.synchronizedList(new ArrayList<>());
             if (CollectionUtils.isNotEmpty(list)) {
                 List<String> xids = list.stream().flatMap(ll -> ll.stream()).collect(Collectors.toList());
                 xids.parallelStream().forEach(xid -> {
-                    GlobalSession globalSession = this.readSession(xid, true);
+                    GlobalSession globalSession = this.readSession(xid, withBranchSessions);
                     if (globalSession != null) {
+                        globalSession.setLazyLoadBranch(!withBranchSessions);
```

Comment:
LazyLoadBranch should be set inside the readSession method

Reply:
done